server/controllers/UsersController.js

```diff
@@ -0,0 +1,218 @@
+        const checkPassword = user.validatePassword(
+          req.body.password,
+          returningUser.password
+        );
+        if (!checkPassword) {
+          return res.status(401).send({
+            message: 'Invalid password'
+          });
+        }
+        const token = user.generateJWT(returningUser.id, returningUser.role);
+        return res.status(200).send({
+          success: true,
+          message: 'Login successful! :)',
+          token,
+          userDetails: {
+            name: returningUser.fullName,
+            id: returningUser.id,
+            role: returningUser.role
+          }
+        });
+      })
+      .catch(err => res.status(401).send(err));
+  }
+
+  /**
+   * @description
+   * @static
+   * @param {any} req
+   * @param {any} res
+   * @returns {object} response
+   * @memberof UsersController
+   */
+  static createUser(req, res) {
+    if (req.headers.authorization) {
+      if (req.decoded.role === 'user') {
+        return res.status(401).send({
+          success: false,
+          message: 'Unathorized access! Only an Admin can create a user'
+        });
+      }
+      return UsersController.registerUser(req, res);
+    }
+    return res.status(400).send({
+      message: 'Please set token in the header'
+    });
+  }
+
+  /**
+   * @description
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} response
+   * @memberof UsersController
+   */
+  static getUsers(req, res) {
+    if (req.decoded.role !== 'admin') {
+      return res.status(401).send({
+        message: 'Unauthorized access! All users can only be viewed by an admin'
+      });
+    }
+    User.findAll()
+      .then(users => res.status(200).send(users))
+      .catch(err => res.status(401).send(err));
+  }
+
+  /**
+   * @description
+   * @static
+   * @param {any} req
+   * @param {any} res
+   * @returns {object} response
+   * @memberof UsersController
+   */
+  static updateUser(req, res) {
+    if (!Number.isInteger(Number(req.params.id))) {
+      return res.status(400).send({
+        message: 'Invalid User ID'
+      });
+    }
+    if (req.body.password) {
+      const user = new User();
+      req.body.password = user.generateHash(req.body.password);
+    }
+    return User.findById(req.params.id)
+      .then((user) => {
+        if (!user) {
+          return res.status(400).send({
+            message: 'Sorry, the user does not exist!'
+          });
+        }
+        if (Number(req.decoded.id) !== Number(req.params.id)) {
+          return res.status(401).send({
+            message: 'Unauthorized Access'
+          });
+        }
+        if (req.body.role) {
+          if (req.decoded.role !== 'admin') {
+            return res.status(401).send({
+              message: 'Unauthorized Access, Only Admin can Update Role'
+            });
+          }
+        }
+        return user
+          .update(req.body, { fields: Object.keys(req.body) })
```

Comment:
I am not sure `{ fields: Object.keys(req.body) }` will be needed except you want to go the extra mile to specify the fields that the user can update which is a good thing.